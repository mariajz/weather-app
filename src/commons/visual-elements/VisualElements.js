import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import {
    StyledBackgroundImage,
    StyledIcon,
    StyledImage,
    StyledGradientBar,
    Slider,
} from './VisualElements.style';
import { icons } from '../../../assets/icons';

export const BackgroundImage = () => {
    return (
        <StyledBackgroundImage
            source={require('../../../assets/images/background.png')}
            blurRadius={100}
            resizeMode="stretch"
            testID="bgImage"
        />
    );
};

export const Theme = {
    BackgroundWhite: opacity => `rgba(255,255,255, ${opacity})`,
};

export const CustomIcon = ({ iconName, size, tintColor, onPress }) => {
    const path = icons[iconName];
    return (
        <If condition={onPress}>
            <TouchableOpacity onPress={onPress}>
                <StyledIcon
                    testID="icon"
                    source={path}
                    size={size}
                    tintColor={tintColor}
                />
            </TouchableOpacity>
            <Else />
            <StyledIcon
                testID="icon"
                source={path}
                size={size}
                tintColor={tintColor}
            />
        </If>
    );
};

export const CustomImage = ({ path, width, height }) => {
    return (
        <StyledImage
            testID="image"
            source={path}
            width={width}
            height={height}
        />
    );
};

export const GradientColorBar = ({ minValue, maxValue, sliderPosition }) => {
    const range = maxValue - minValue;
    const normalizedValue = (sliderPosition - minValue) / range;
    let calculatedLeft = normalizedValue * 100;
    calculatedLeft = calculatedLeft > 100 ? 100 : calculatedLeft;
    calculatedLeft = calculatedLeft < 0 ? 0 : calculatedLeft;

    return (
        <View>
            <StyledGradientBar />
            <Slider left={calculatedLeft} />
        </View>
    );
};
