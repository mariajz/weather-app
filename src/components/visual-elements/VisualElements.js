import React from 'react';
import { StyledBackgroundImage } from './VisualElements.style';

export const BackgroundImage = () => {
    return (
        <StyledBackgroundImage
            source={require('../../../assets/background-2.png')}
            blurRadius={100}
            resizeMode="stretch"
        />
    );
};
