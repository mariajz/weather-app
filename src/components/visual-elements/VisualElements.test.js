import { render } from '@testing-library/react-native';
import React from 'react';
import { BackgroundImage } from './VisualElements';

describe('AppWrapper', () => {
    it('should render BackgroundImage', () => {
        const container = render(<BackgroundImage />);

        expect(container).toMatchSnapshot();
    });
});
