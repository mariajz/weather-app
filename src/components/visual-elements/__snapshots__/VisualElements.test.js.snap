// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppWrapper should render BackgroundImage 1`] = `
<Image
  blurRadius={100}
  resizeMode="stretch"
  source={
    {
      "testUri": "../../../assets/background-2.png",
    }
  }
  style={
    {
      "position": "absolute",
    }
  }
/>
`;
